src/applications/personalization/dashboard/tests/helpers.unit.spec.js

```diff
@@ -143,9 +147,119 @@ describe('profile helpers:', () => {
+      metadata: {
+        version: 3,
+        returnUrl: '/review-and-submit',
+        savedAt: 1557507430028,
+        expiresAt: 1562691437,
+        lastUpdated: 1557507437,
+      },
+      lastUpdated: 1557507437,
+    };
+    const formB = {
+      form: '1010ez',
+      metadata: {
+        version: 6,
+        returnUrl: '/military-service/documents',
+        savedAt: 1558027285255,
+        expiresAt: 1563211286,
+        lastUpdated: 1558027286,
+      },
+      lastUpdated: 1558027286,
+    };
+    const formC = {
+      form: '22-1995-STEM',
+      metadata: {
+        version: 1,
+        returnUrl: '/personal-information/contact-information',
+        savedAt: 1558440074236,
+        expiresAt: 1563624074,
+        lastUpdated: 1558440074,
+      },
+      lastUpdated: 1558440074,
+    };
+    const invalidExpiresAt = {
+      metadata: {
+        expiresAt: 'not a number',
+      },
+    };
+    const notAForm = {
+      veteranStatus: {
+        status: 'OK',
+        isVeteran: true,
+        servedInMilitary: true,
+      },
+    };
+    describe('unhappy path', () => {
+      it('should throw an error when no args are passed', () => {
+        expect(() => sipFormSorter()).to.throw(TypeError, 'plain object');
+      });
+      it('should throw an error when a single invalid object is passed', () => {
+        expect(() => sipFormSorter(notAForm)).to.throw(
+          TypeError,
+          'metadata.expiresAt',
+        );
+      });
+      it('should throw an error when a single valid form is passed', () => {
+        expect(() => sipFormSorter(formA)).to.throw(TypeError, 'plain object');
+      });
+      it('should throw an error when the second arg is a string', () => {
+        expect(() => sipFormSorter(formA, 'not a form')).to.throw(
+          TypeError,
+          'plain object',
+        );
+      });
+      it('should throw an error when the second arg is an array', () => {
+        expect(() => sipFormSorter(formA, [])).to.throw(
+          TypeError,
+          'plain object',
+        );
+      });
+      it('should throw an error when the second arg is a number', () => {
+        expect(() => sipFormSorter(formA, 42)).to.throw(
+          TypeError,
+          'plain object',
+        );
+      });
+      it('should throw an error when the second arg is an invalid object', () => {
+        expect(() => sipFormSorter(formA, notAForm)).to.throw(
+          TypeError,
+          'metadata.expiresAt',
+        );
+      });
+      it('should throw an error when the second arg is an empty object', () => {
+        expect(() => sipFormSorter(formA, {})).to.throw(
+          TypeError,
+          'metadata.expiresAt',
+        );
+      });
+      it('should throw an error when an arg has an invalid value for `metadata.expiresAt`', () => {
+        expect(() => sipFormSorter(formA, invalidExpiresAt)).to.throw(
+          TypeError,
+          'metadata.expiresAt',
+        );
+      });
+    });
```

Comment:
I haven't explored the data very closely, but these are all cases that could happen while sorting `profile.savedForms`?

Reply:
I was being overly cautious of the sorting function itself, since it's just a public function that could be used anywhere by anyone. If you were actually using `sipFormSorter` as a callback to Array.sort() some of these cases wouldn't come up. I was mostly concerned about the cases where the `metadata.expiresAt` is not defined on the object passed to this function and wanted to make sure that case was handled decently.